### PR TITLE
Show helm release revisions in history tab

### PR DESCRIPTION
**Pull request by divyanshiGupta**
**Fixes:** https://issues.redhat.com/browse/ODC-3158

**Analysis / Root cause:**
Add history tab on helm release details page and show helm release revisions

**Solution Description:**
This PR adds history tab on `HelmReleaseDetailsPage`. Fetches helm release revisions from API endpoint `/api/helm/release/history?ns=${namespace}&name=${name}` and shows it as a list.

**Screen shots / Gifs for design review:**
Screenshot from 2020-03-20 17-24-18

Updated
Screenshot from 2020-03-23 20-10-07

**Test setup:**
The UI depends on APIs being added by #4580. Need to build the backend on top of that PR in order to test the API.

To-do
- [x] Add unit tests

### CODE REVIEW

**Comment by rohitkrai03** [comment]
@divyanshiGupta Some unit tests are failing.

**Comment by christianvogt** [comment]
cc @siamaksade @openshift/team-devconsole-ux

File: frontend/packages/dev-console/src/components/custom-resource-list/custom-resource-list-types.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export enum ListFilterType {
+  Row = 'row',
+  Text = 'text',
+}
+
+export interface ListProps {
+  items: Promise<any>;
+  queryArg: string;
+  rowFilters: any;
+  resourceRow: React.FC;
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/custom-resource-list/custom-resource-list-types.ts]: Could you add prop types here?
```suggestion
  resourceRow: React.ComponentType<...>;
```

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx [hunk 2]
```diff
@@ -60,6 +65,20 @@ const HelmReleaseDetailsPage: React.FC<HelmReleaseDetailsPageProps> = ({ secret,
...
+            name: 'History',
+            component: () => (
+              <CustomResourceList
+                items={getHelmReleaseRevisions(namespace, helmReleaseName)}
+                queryArg="rowFilter-helm-release-status"
+                rowFilters={helmRowFilters}
+                getFilteredItems={getFilteredItems}
+                resourceRow={HelmReleaseRow}
+                resourceHeader={HelmReleaseHeader}
+              />
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx]: Make a separate component: `HelmReleaseHistory`

**Comment by debsmita1** [comment]
The below error occured while sorting the name column
history

File: frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  React.useEffect(() => {
+    let ignore = false;
+
+    const queryArgument = getQueryArgument(queryArg);
+    const activeFilters = queryArgument?.split(',');
+
+    const fetchListItems = async () => {
+      let newListItems: any;
+      try {
+        newListItems = await items;
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx]: I would rather rename this to be `fetchCustomResources` and make it a function that returns a promise to make it more clear that this is used to fetch the resources.

File: frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    fetchListItems();
+
+    return () => {
+      ignore = true;
+    };
+  }, [dependentResource, items, queryArg, rowFilters, getFilteredItems]);
+
+  const applyRowFilter = React.useCallback(
+    (filter) => {
+      const filteredItems = getFilteredItems(listItems, CustomResourceListFilterType.Row, filter);
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx]: This makes it very specific to the component. I would prefer if the parent just sent `rowFilterReducer`.

File: frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      setFilteredListItems(filteredItems);
+    },
+    [listItems, getFilteredItems],
+  );
+
+  const applyTextFilter = React.useCallback(
+    (filter) => {
+      const filteredItems = getFilteredItems(listItems, CustomResourceListFilterType.Text, filter);
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx]: Same as above. Would be better to just default this to `fuzzy(items, item.name)` and give an ability to provide `filterBy` prop from parent.

File: frontend/packages/dev-console/src/components/helm/HelmReleasePage.tsx [hunk 6]
```diff
@@ -33,7 +33,7 @@ export const HelmReleasePage: React.FC<HelmReleasePageProps> = (props) => {
       </Helmet>
       <PageHeading title="Helm Releases" />
       <Firehose resources={resources}>
-        <HelmReleaseList namespace={namespace} />
+        <HelmReleases namespace={namespace} />
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/helm/HelmReleasePage.tsx]:
```suggestion
        <HelmReleaseList namespace={namespace} />
```

**Comment by divyanshiGupta** [comment]
I had to create a new PR #4817 due to some local setup issue. Closing this one.

**Pull request closed without merging**